Updates

D-sha opened the pull request:
Work in progress. Not complete yet. I would appreciate some feedback thankyou

Commit:
update fred to current (1467)
hopefully i dident screw this up

Commit:
Removed sid links for gwt, removed from repo - no longer maintained
Changed Bouncy Castle links jessie and sid have been updated to 1.49 but 151 is now required
Added bouncycastle.org

Thynix commented:
Looks fine to me. Do you have ideas for what we can do about requiring dependencies more recent than distros tend to ship?

RdrOfTheSt0rm commented:
My work on this has unfortunately stalled as i have never successfully built a deb from the original freenet/debian source. I am getting an indexoutofbounds exception while compiling the js in the build-freenet-daemon. I will have to learn a bit more about freenet to troubleshoot this.

As for the dependencies, i guess we can require that they are there before freenet is installed, work with the upstream to get the packages accepted or ship with the deb package?? I am not sure which method is best. I will investigate this further.
This project has been a steep learning curve for me but im enjoying it.

Does the build script work? Could somebody let me know if it does so i know if the problem is at my end?

infinity0 commented:
I believe a major problem is that GWT is no longer in debian.

RdrOfTheSt0rm commented:
It should still work with the squeeze version correct? and thats only required for building right?

infinity0 commented:
No, it would be required for running too. Java JARs are dynamically loaded at runtime.

Thynix commented:
Is JWT for anything other than the image loading progress placeholders?

On Tue, Jan 20, 2015, 6:26 AM Ximin Luo <notifications@github.com> wrote:

> No, it would be required for running too. Java JARs are dynamically loaded
> at runtime.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freenet/debian/pull/4#issuecomment-70640015>.
>

Thynix commented:
I ask because if it's only for the image loading placeholders perhaps that feature can be removed in order to get the Debian package working? Maybe that's not something people would appreciate... We could have a development branch that removes things from / modifies Fred until a Debian package is practical, at the very least.

D-sha committed:
freenetdaemon: update fred to 1470, remove -x from git clean, remove gwt from depends

Fred is updated to current master (1470)
git clean -fdx was removing bcprov from freenet-daemon/fred/lib
gwt is no longer in debian repos this will have to be done manually

D-sha committed:
Modify debian/rules build file

Ant build file changed in debian/rules from build-clean.xml to build.xml
Added -Dtest.skip=true to save build time when testing

D-sha committed:
Update debian/freenet.override.properties

Update: bouncycastle file name
remove /usr/share/java from lib folder as
this is already in fred/build-clean.xml
remove bouncycastle from test.jars

D-sha committed:
Update freenet-daemon.docs debian/control

Change README to README.md in freenet-daemon.docs
Removed lines from control file that caused complaints from dpkg-build[ackage

D-sha committed:
Use contrib v29

Freenet uses freenet-ext.jar v29 so we will too.

D-sha committed:
Add third party librarys, other small misc changes

Add gwt too debian/ directory
Download bouncycastle and add to package

D-sha committed:
Move bouncycastle download out of update

build-freenet-daemon.sh placed bc download under update submodule
so build script will just work without args

D-sha committed:
Update README's and TODO

D-sha committed:
Use https for build-freenet-daemon.sh downloads

D-sha committed:
Fix incorrect paths

D-sha committed:
Update seednodes

D-sha committed:
Fixed .gitmodules contrib url, Fixed wget downloads in the build script
